### PR TITLE
Updates due to JuliaDiff/TaylorSeries.jl#361

### PR DESCRIPTION
**Pull request by PerezHz**
This PR contains updates which go together with JuliaDiff/TaylorSeries.jl#361

EDIT: this PR depends also on JuliaDiff/TaylorSeries.jl#364

### CODE REVIEW

**Comment by PerezHz** [comment]
Tests were actually passing modulo coverage, except for the last commit, but cannot reproduce the issue locally, so... I'm re-running tests and see if it's maybe a CI issue

**Comment by PerezHz** [comment]
After re-running tests, they are passing except for Julia 1.6 on the macos platform... does look like a CI issue?

**Comment by lbenet** [comment]
> After re-running tests, they are passing except for Julia 1.6 on the macos platform... does look like a CI issue?

Also windows in Julia 1.6 is failing. I doubt it is a CI issue, it should be deterministic, right?

**Comment by PerezHz** [comment]
> Also windows in Julia 1.6 is failing. I doubt it is a CI issue, it should be deterministic, right?

So you're saying maybe it's an initialization issue?

**Comment by PerezHz** [comment]
Windows and MacOS failures on Julia 1.6 look pretty consistent, whereas all the other versions and platforms tests passing also look consistent. Probably it's then a `sqrt` issue which is somehow only visible on Julia 1.6 on Windows and MacOS but not on Ubuntu?

**Comment by PerezHz** [comment]
Update: managed to reproduce the error on julia 1.6 (macos)

**Comment by PerezHz** [comment]
Pushed a fix to TaylorSeries...

**Comment by PerezHz** [comment]
... et voilà!

**Comment by PerezHz** [comment]
In a nutshell, since some allocations were incurred in the preamble of parsed `jetcoeffs!` methods, in one of the last commits I removed the preamble in favor of using mutating methods for 0-th order, modifying the recursion loop accordingly. For power operations such as `x = r^3` the preamble did a float conversion `x = Taylor1(constant_term(r) ^ float(constant_term(3)), order)`. The equivalent conversion in `pow!` was missing, and hence different results were obtained by parsed and non-parsed `jetcoeffs!` methods for Julia 1.6 (and my guess is that this issue somehow did not make it to the official Linux x64 binaries, so the issue did not appear there).

**Comment by lbenet** [comment]
> In a nutshell, since some allocations were incurred in the preamble of parsed `jetcoeffs!` methods, in one of the last commits I removed the preamble in favor of using mutating methods for 0-th order, modifying the recursion loop accordingly. For power operations such as `x = r^3` the preamble did a float conversion `x = Taylor1(constant_term(r) ^ float(constant_term(3)), order)`. The equivalent conversion in `pow!` was missing, and hence different results were obtained by parsed and non-parsed `jetcoeffs!` methods for Julia 1.6 (and my guess is that this issue somehow did not make it to the official Linux x64 binaries, so the issue did not appear there).

Thanks for checking and solving this!

**Comment by lbenet** [comment]
Registration of Taylor series v0.18.0 is on its way... 10-15 minutes are required :smile:

**Comment by lbenet** [comment]
v0.18.0 is released!

**Comment by PerezHz** [comment]
> v0.18.0 is released!

Many thanks! I'll re-trigger tests here!

File: src/TaylorIntegration.jl [hunk 1]
```diff
@@ -29,20 +29,20 @@ function __init__()
 end
 
 
-@inline function diffeq!(a::Taylor1{T}, b::Taylor1{T}, k::Int) where {T<:TaylorSeries.NumberNotSeries}
+@inline function ode!(a::Taylor1{T}, b::Taylor1{T}, k::Int) where {T<:TaylorSeries.NumberNotSeries}
```
Comment (lbenet) [inline src/TaylorIntegration.jl:32]: What about `solcoeff!` or something alike?

**Comment by PerezHz** [comment]
With TaylorSeries v0.18.0 released this is now ready for review

File: src/TaylorIntegration.jl (hunk 1 above)
Comment (PerezHz) [reply src/TaylorIntegration.jl]: Done; thanks for the suggestion!

**Comment by lbenet** [comment]
> With TaylorSeries v0.18.0 released this is now ready for review

I'll try to review it this afternoon... sorry for the delay.

**Review by lbenet — APPROVED** [review]
Thanks a lot @PerezHz. My only suggestion is to bump a new (minor or patch?) version. Aside from that, LGTM!

**Comment by PerezHz** [comment]
> Thanks a lot @PerezHz. My only suggestion is to bump a new (minor or patch?) version. Aside from that, LGTM!

Alright, bumped the patch version and will merge afterwards, thanks!

**Pull request merged**